### ISSUE TITLE
Jayrock.dll and JayrockJson.dll missing details in file properties

### ISSUE THREAD

**Issue by PellaAndroid** [kee-org/KeeFox#671]
OS: Windows 10 Home 64-bit
browser: Waterfox 46.0.1
KeePass: 2.33
KeeFox: 1.6.0

After updating KeeFox to 1.6, I kept seeing the "Launch KeePass" button in Waterfox. I uninstalled KeeFox and reinstalled with no change. KeePassRPC did not show in the list of plugins within KeePass.

I closed KeePass and restarted it, and I got this error:
The following plugin is incompatible with the current KeePass version:
C:\Users<username>\AppData\Local\KeePass\PluginCache\dCflSn65gzKzNziw5Z1G\KeePassRPC.dll

Have a look at the plugin's website for an appropriate version.

Could not load file or assembly 'Jayrock, Version=0.9.12915.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. The system cannot find the file specified.
KeePassRPC
    at KeePassRPC.KeePassRPCExt.Initialize(!PluginHost host)
    at KeePass.Plugins.PluginManager.LoadPlugins(FileInfo[] vFiles, String strTypeName, String strDisplayFilePath, Boolean bSkipCacheFiles)
Boolean Initialize(KeePass.Plugins.!PluginHost)

Inside the directory listed are five files:
- DomainPublicSuffix.dll
- Fleck2.dll
- Jayrock.dll
- JayrockJson.dll
- KeePassRPC.dll

Three of these files are just fine. However, when examining the Properties for Jayrock and JayrockJson, on the Details tab, I observed that the only fields with information were:
- Type
- Size
- Data modified

Both files listed the all the same fields as the three good files, with the exception of "Original filename", yet the six other important fields all were blank.

Bottom line: Although KeeFox installed all the required files, the missing information on the Jayrock and JayrockJson files prevented KeePass from recognizing them as the correct files, which in turn prevented KeeFox from loading at all.

Inside a different folder of PluginCache, I found Jayrock and JayrockJson files with all Details fields completed as expected, showing Product version 0.9.12915.0. I used these two files to replace the files with missing details. The next time I started KeePass, KeeFox prompted me to enter the short password. Everything now works perfectly.

**Comment by luckyrat** [comment]
> the missing information on the Jayrock and JayrockJson files prevented KeePass from recognizing them as the correct files

Are you certain this was the reason they could not be loaded? I'm wondering whether the lack of information in the details tab is simply a symptom of a bigger problem such as the 2 files being corrupted on disk, perhaps as a result of a 3rd party security application bug, a KeePass bug or even a hardware fault like an unreliable disk drive or RAM. One clue (although not definitive) would be if the size of a broken and working file was significantly different.

**Comment by PellaAndroid** [comment]
Ah, good call. I intended to mention that, as well.

Obviously, the files got corrupted somehow. However, the file sizes of 
the bad and good files are identical. That is the primary reason I 
stated that the missing information was the cause. It's clearly not the 
root cause, though. How did that information get stripped?

I have attached the two bad files so you can analyze them yourself. I 
have no way to tell whether they're good inside or not. ;-)

==Rich M++
